src/transformers/utils/model_parallel_utils.py

```diff
@@ -12,10 +12,186 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import copy
+import importlib
+from collections import defaultdict
 from math import ceil
 
 
+class TPInfo(object):
```

Comment:
We need a docstring here explaining what this class does.

Also, the name is unclear to someone who is not familiar with all the model parallelism jargon. Let's expand to `TensorParallelismInfo` as we usually use descriptive names in Transformers.